sootty/visualizer.py

```diff
@@ -225,6 +228,8 @@ def _wire_to_svg(self, wire, left, top, start, length):
             }
         )
         for index in range(start, start + length):
+            if vector_radix != 10 and wire.width() != 1 and type(wire[index]) == int:  # VECTOR CHANGE DATA (no dup)
+                wire[index] = dec2anybase(wire[index], vector_radix, wire.width())
```

Comment:
The value of `wire[]` shouldn't be mutated in this function. Since we need to modify how the value is printed it should be done in the call to `_value_to_svg` (line 235).

Reply:
The work has been moved to `_value_to_svg`.